crates/control/src/search_suggestor.rs

```
@@ -106,11 +108,30 @@ impl SearchSuggestor {
                 self.heatmap[rule_ball_hypothesis] = 1.0;
             }
         }
-
-        self.heatmap.map *= 1.0 - context.search_suggestor_configuration.heatmap_decay_factor;
+        let kernel = create_kernel(
+            context
+                .search_suggestor_configuration
+                .heatmap_convolution_kernel_weight,
+        );
+        if let Ok(convolved_heatmap) =
```

When does the convolution return an error? I can imagine this happens only for when the kernel is larger than the heatmap. Right now you are silently ignoring the error. If for some reason this call would only generate Errors, we would not notice it. I though of using `.wrap_err("heatmap convolution failed")?`

---

this should be better now